src/Microsoft.Framework.ConfigurationModel/Sources/CommandLineConfigurationSource.cs

```diff
@@ -33,9 +34,9 @@ public override void Load()
             string key, value;
             var argIndex = 0;
 
-            while (argIndex < Args.Length)
+            while (argIndex < Args.Count())
```

Comment:
Oh nevermind I see we need to look at the "next" arg here. perhaps that means that `IEnumerable<T>` is the wrong contract. Why not use ICollection or something stronger?